### DISCUSSION TITLE
the tag generation feature can only read english parts? how to fix it?

### DISCUSSION THREAD

**Discussion by panAtGitHub** [ObsidianToAnki/Obsidian_to_Anki#550] [Q&A] [answered]
你好，我在使用 obsidian to anki 的标签生成功能时，发现其只能读取英文部分，不能读取中文部分，不知这是否是个 bug，可否修复下，谢谢。
PS：我的试验过程如下文所示：
Hello, I've noticed when using the tag generation feature from Obsidian to Anki that it can only read the English part and fails to read the Chinese part. I'm not sure if this is a bug, but could it be fixed, please? Thank you.
P.S. My experimental process is as described in the following text:

#### 试验 1：纯英文标签
结果：anki 可以正常读取英文标签
Results: Anki can properly read English tags.
Tags: #note  #TARGET #important 

#### 试验 2：纯中文标签
结果：不行，不能读取中文标签
Results: No, it cannot read Chinese tags.
Tags: #目标 #笔记 #重要 

#### 试验 3：英文+中文组合标签
结果：anki 读取英文部分，不能读取中文部分
Results: Anki reads the English part, cannot read the Chinese part.
Tags: #note笔记  #TARGET目标 #important重要 

#### 试验 4：中文+英文组合标签
结果：anki 不能读取相关标签，即中文在前时不能进行识别
Results: Anki cannot recognize the related tags when Chinese comes first.
Tags: #目标target #笔记note #重要important

#### 试验 5：试验一、二、三的组合排列
结果：混合标签情况下，还是试验一、二的标签可以正常读取，试验三的标签样式不得行
Results: In the case of mixed tags, only the tags from experiments 1 and 2 can be properly read; the tag style from experiment 3 does not work.
Tags: #note #note笔记 #笔记note  #笔记 

#### 试验 6：不用 Tags，用其他词
结果：正常生成英文标签
Results: English tags are generated normally.
T： #note #note笔记 #笔记note 
标签： #note笔记  #TARGET目标 #important重要 

#### 试验结论
推荐使用：英文、或英文+中文组合标签，即试验 1 与 3 两种形式。
Recommendation: Use English or English + Chinese combination tags, namely the forms from experiments 1 and 3.
![image.png|400](https://ppicbed.oss-cn-hangzhou.aliyuncs.com/img/202403181110952.png)

**Comment by envico801** [comment, marked as answer]
Hi @panAtGitHub , I left you a possible solution in #551
